src/main/java/de/maxhenkel/reap/TreeEvents.java

```diff
@@ -59,97 +63,104 @@ public static boolean canHarvest(BlockPos pos, Player player, Level world, ItemS
             return false;
         }
 
-        if (!isLog(world, pos)) {
-            return false;
-        }
+        return true;
+    }
 
-        if (!isGround(world, pos.below())) {
-            return false;
+    private void destroyTree(Player player, Level world, BlockPos pos) {
+        LinkedList<BlockPos> connectedLogs = scanForTree(world, pos);
+        for (BlockPos log : connectedLogs) {
+            destroy(world, player, log);
         }
+    }
 
-        BlockState state = world.getBlockState(pos);
-        if (state.getProperties().stream().anyMatch(p -> p.equals(RotatedPillarBlock.AXIS))) {
-            if (!state.getValue(RotatedPillarBlock.AXIS).equals(Direction.Axis.Y)) {
-                return false;
-            }
+    private static void destroy(Level world, Player player, BlockPos pos) {
+        ItemStack heldItem = player.getMainHandItem();
+        if (heldItem != null) {
+            heldItem.getItem().mineBlock(heldItem, world, world.getBlockState(pos), pos, player);
+            world.destroyBlock(pos, true);
+            player.causeFoodExhaustion(0.025F);
         }
-
-        return true;
     }
 
-    private static void destroyTree(Player player, Level world, BlockPos pos, ItemStack heldItem) {
-        List<BlockPos> connectedLogs = getConnectedLogs(world, pos);
-
-        for (BlockPos logPos : connectedLogs) {
-            destroy(world, player, logPos, heldItem);
+    // tree detector
+    public static LinkedList<BlockPos> scanForTree(final LevelReader level, final BlockPos start) {
+        final BlockState blockState = level.getBlockState(start);
+        if (!blockState.is(BlockTags.LOGS)) {
```

Comment:
Why did you remove the ability to configure which type of logs can be harvested?

Reply:
Re-added config for custom trees back